components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/main/java/org/wso2/carbon/apimgt/webapp/publisher/config/Profiles.java

```diff
@@ -0,0 +1,57 @@
+
+package org.wso2.carbon.apimgt.webapp.publisher.config;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlType;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * <p>Java class for Profiles complex type.
+ * 
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ * 
+ * <pre>
+ * &lt;complexType name="Profiles">
+ *   &lt;complexContent>
+ *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
+ *       &lt;sequence>
+ *         &lt;element name="Profile" maxOccurs="unbounded" minOccurs="0">
+ *           &lt;simpleType>
+ *             &lt;restriction base="{http://www.w3.org/2001/XMLSchema}string">
+ *               &lt;enumeration value="default"/>
+ *             &lt;/restriction>
+ *           &lt;/simpleType>
+ *         &lt;/element>
+ *       &lt;/sequence>
+ *     &lt;/restriction>
+ *   &lt;/complexContent>
+ * &lt;/complexType>
+ * </pre>
+ * 
+ * 
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "Profiles", propOrder = {
+    "profile"
+})
+public class Profiles {
+
+    @XmlElement(name = "Profile")
+    protected List<String> profile;
+
+    /**
+     * Gets the value of the profile property.
+     *
+     */
+    public List<String> getProfile() {
+        if (profile == null) {
+            profile = new ArrayList<String>();
```

Comment:
Make these variables get initialized when the class is loaded so you can then get rid of the null checks. That said, I wouldn't drop the fact that tactics similar to what you've used are generally employed when initializing heavy-weight objects thereby allowing the code to lazy-initialize objects, which is not the case in this particular instance.